upf/upf.c

```diff
@@ -201,17 +201,23 @@ vnet_upf_upip_add_del (ip4_address_t * ip4, ip6_address_t * ip6,
 {
   upf_main_t *gtm = &upf_main;
   upf_upip_res_t *ip_res;
+  u8 *key = 0;
   uword *p;
-  u8 *unformat_nwi;
 
   if (!name)
     {
       upf_debug ("NWI is mandatory for GTPU endpoint");
       return -1;
     }
 
-  unformat_nwi = format (0, "%U", format_dns_labels, name);
-  p = hash_get_mem (gtm->upip_res_index_by_nwi, unformat_nwi);
+  /*
+   * TODO: For some reason pure usage of binary vector breaks
+   * hash lookup by key. Converting to string works like a
+   * workaround, but this has to be investigated and fixed.
+   */
+  name = format (0, "%s", name);
```

Comment:
I'm still waiting to see a good explanation of what exactly is not working with the vector encoding.